core/src/test/java/org/bitcoinj/wallet/DeterministicKeyChainTest.java

```diff
@@ -470,6 +475,180 @@ public void watchingChainAccountOne() throws UnreadableWalletException {
         assertEquals(key4.getPubKeyPoint(), rekey4.getPubKeyPoint());
     }
 
+    @Test
+    public void spendingChain() throws UnreadableWalletException {
+        Utils.setMockClock();
+        DeterministicKey key1 = chain.getKey(KeyChain.KeyPurpose.RECEIVE_FUNDS);
+        DeterministicKey key2 = chain.getKey(KeyChain.KeyPurpose.RECEIVE_FUNDS);
+        DeterministicKey key3 = chain.getKey(KeyChain.KeyPurpose.CHANGE);
+        DeterministicKey key4 = chain.getKey(KeyChain.KeyPurpose.CHANGE);
+
+        NetworkParameters params = MainNetParams.get();
+        DeterministicKey watchingKey = chain.getWatchingKey();
+        final String prv58 = watchingKey.serializePrivB58(params);
+        assertEquals("xprv9vL4k9HYXonmvqGSUrRM6wGEmx3ruGTXi4JxHRiwEvwDwYmTocPbQNpjN89gpqPrFofmfvALwgnNFBCH2grse1YDf8ERAwgdvbjRtoMfsbV", prv58);
+        watchingKey = DeterministicKey.deserializeB58(null, prv58, params);
+        watchingKey.setCreationTimeSeconds(100000);
+        chain = DeterministicKeyChain.spend(watchingKey);
+        assertEquals(100000, chain.getEarliestKeyCreationTime());
+        chain.setLookaheadSize(10);
+        chain.maybeLookAhead();
+
+        assertEquals(key1.getPubKeyPoint(), chain.getKey(KeyChain.KeyPurpose.RECEIVE_FUNDS).getPubKeyPoint());
+        assertEquals(key2.getPubKeyPoint(), chain.getKey(KeyChain.KeyPurpose.RECEIVE_FUNDS).getPubKeyPoint());
+        final DeterministicKey key = chain.getKey(KeyChain.KeyPurpose.CHANGE);
+        assertEquals(key3.getPubKeyPoint(), key.getPubKeyPoint());
+        try {
+            // We can sign with a key from a spending chain.
```

Comment:
this comment could go into the `catch` logic:

`fail("Keys from spending chain should be able to sign");`